cloudregister/smt.py

```diff
@@ -201,8 +202,13 @@ def set_protocol(self, protocol):
     # --------------------------------------------------------------------
     def write_cert(self, target_dir):
         """Write the certificate to the given directory"""
-        logging.info('Writing SMT rootCA: %s' % target_dir)
         cert = self.get_cert()
+        if not cert:
+            # the cert is invalid or
+            # the request to fetch the cert failed
+            return 0
```

Comment:
Hmm for `import_smt_cert()` we return `None` on failure, we should probably be consistent with the return type on failure if at all possible, which in this case appears to be possible.